tools/evmc/main.cpp

```diff
@@ -10,25 +10,27 @@
 
 namespace
 {
-/// Returns the input str if already valid hex string. Otherwise, interprets the str as a file
-/// name and loads the file content.
+/// If the argument starts with @ returns the content of the file at the path following @.
+/// Otherwise, returns the argument.
 /// @todo The file content is expected to be a hex string but not validated.
 std::string load_hex(const std::string& str)
 {
-    if (evmc::validate_hex(str))
-        return str;
+    if (str[0] == '@')  // The argument is file path.
+    {
+        std::ifstream file{str.c_str() + 1};
+        return {std::istreambuf_iterator<char>{file}, std::istreambuf_iterator<char>{}};
+    }
 
-    // Must be a file path.
-    std::ifstream file{str};
-    return {std::istreambuf_iterator<char>{file}, std::istreambuf_iterator<char>{}};
+    return str;
 }
 
-struct HexValidator : public CLI::Validator
+struct HexOrFileValidator : public CLI::Validator
 {
-    HexValidator() : CLI::Validator{"HEX"}
+    HexOrFileValidator() : CLI::Validator{"HEX|@FILE"}
     {
-        name_ = "HEX";
         func_ = [](const std::string& str) -> std::string {
+            if (!str.empty() && str[0] == '@')
+                return CLI::ExistingFile(str.substr(1));
```

Comment:
Shouldn't this try to load & verify the content?

Reply:
It could but it feels too much. I don't know any CLI library doing so. And you would need to load the file twice.

Reply:
Since the files are usually small, the overhead may be worth it here for easy-of-use (and not relying on exceptions).